migration/v2.27/apply/00-migrate-loadbalancers.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+: "${CK8S_CONFIG_PATH:?Missing CK8S_CONFIG_PATH}"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+openstack_upcloud_dir="${HERE}/../../../kubespray/contrib/terraform/upcloud"
+
+clusters=()
+
+if [[ "${CK8S_CLUSTER}" == "both" ]]; then
+  clusters+=("sc")
+  clusters+=("wc")
+else
+  clusters+=("${CK8S_CLUSTER}")
+fi
+
+for cluster in "${clusters[@]}"; do
+
+  export TF_VAR_inventory_file="${CK8S_CONFIG_PATH}/${cluster}-config/inventory.ini"
+
+  log_info "Enter the new name for the loadbalancer in the ${cluster} cluster:"
+  read -r lb_name
+  if [[ -z "${lb_name}" ]]; then
+    log_fatal "No loadbalancer was given!"
+  fi
+
+  if [[ -n $(jq -r '.resources[] | select(.type == "upcloud_loadbalancer") | .instances[] | select(.index_key == 0)' "${CK8S_CONFIG_PATH}/${cluster}-config"/terraform.tfstate) ]]; then
+      LB_ID=$(jq -r '.resources[] | select(.type == "upcloud_loadbalancer") | .instances[] |  select(.index_key == 0) | .attributes.id' "${CK8S_CONFIG_PATH}/${cluster}-config"/terraform.tfstate)
+      terraform -chdir="${openstack_upcloud_dir}" import -state="${CK8S_CONFIG_PATH}/${cluster}-config"/terraform.tfstate -var-file "${CK8S_CONFIG_PATH}/${cluster}-config"/cluster.tfvars "module.kubernetes.upcloud_loadbalancer.lb[\"${lb_name}\"]" "${LB_ID}"
+      terraform -chdir="${openstack_upcloud_dir}" state rm -state="${CK8S_CONFIG_PATH}/${cluster}-config"/terraform.tfstate 'module.kubernetes.upcloud_loadbalancer.lb[0]'
+  fi
```

Comment:
You could simplify the reading of the script by doing some minor edits such as:

```suggestion
  terraform_state_file="${CK8S_CONFIG_PATH}/${cluster}-config/terraform.tfstate"
  terraform_var_file="${CK8S_CONFIG_PATH}/${cluster}-config/cluster.tfvars"
  
  lb_state=$(jq -r '.resources[] | select(.type == "upcloud_loadbalancer") | .instances[] | select(.index_key == 0)' "${terraform_state_file}")

  if [[ -n "${lb_state}" ]]; then
      lb_id=$(echo "${lb_state}" | jq -r '.attributes.id')
      terraform -chdir="${openstack_upcloud_dir}" import -state="${terraform_state_file}" -var-file "${terraform_var_file}" "module.kubernetes.upcloud_loadbalancer.lb[\"${lb_name}\"]" "${lb_id}"
      terraform -chdir="${openstack_upcloud_dir}" state rm -state="${terraform_state_file}" 'module.kubernetes.upcloud_loadbalancer.lb[0]'
  fi
```

Reply:
https://github.com/elastisys/compliantkubernetes-kubespray/pull/411/commits/3725b4d0e594546a108c43e8d1effbe1f9ed2930